### PR TITLE
Create IonHashTest

### PR DESCRIPTION
**Pull request by byronlin-bq**
Issue #, if available:
#14 

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ion_hash_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+func TraverseReader(hr HashReader) {
+	for hr.Next() {
+		if hr.Type() != ion.NoType && hr.IsInStruct() {
+			hr.StepIn()
+			TraverseReader(hr)
+			hr.StepOut()
+		}
+	}
```
Comment (justing-bq) [inline ion_hash_test.go:73]: Lets add `assert.NoError(t, hr.Err(), "Something went wrong executing hr.Next()")` after this loop.

File: ion_hash_test.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	_, err = hr.Sum(nil)
+	require.NoError(t, err, "Something went wrong with executing hr.Sum()")
+}
+
+func TraverseReader(hr HashReader) {
+	for hr.Next() {
+		if hr.Type() != ion.NoType && hr.IsInStruct() {
+			hr.StepIn()
+			TraverseReader(hr)
+			hr.StepOut()
```
Comment (justing-bq) [inline ion_hash_test.go:71]: check that StepIn() and StepOut() did not return errors. Likewise with all other unhandled errors.

File: ion_hash_test.go (hunk 1 above)
Comment (byronlin-bq) [reply ion_hash_test.go:73]: Updated

File: ion_hash_test.go (hunk 2 above)
Comment (byronlin-bq) [reply ion_hash_test.go]: Updated

File: ion_hash_test.go (hunk 1 above)
Comment (justing-bq) [reply ion_hash_test.go:73]: Remember that if something goes wrong during hr.Next(), the function will return false and set the internal error inside hr. So we should check hr.Err() after exiting this loop.

File: ion_hash_test.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	}
+}
+
+func Traverse(t *testing.T, reader ion.Reader, provider IonHasherProvider) {
+	hr, err := NewHashReader(reader, provider)
+	require.NoError(t, err, "Something went wrong executing NewHashReader()")
+	TraverseReader(t, hr)
+	_, err = hr.Sum(nil)
+	require.NoError(t, err, "Something went wrong with executing hr.Sum()")
+}
```
Comment (justing-bq) [inline ion_hash_test.go:64]: Style suggestion: I think an empty line before and after TraverseReader(t, hr) would make this a bit more readable.

File: ion_hash_test.go (hunk 1 above)
Comment (byronlin-bq) [reply ion_hash_test.go:73]: Updated

File: ion_hash_test.go (hunk 3 above)
Comment (byronlin-bq) [reply ion_hash_test.go:66]: Addressed.

**Pull request merged**